core/src/main/scala/filodb.core/memstore/PartKeyTantivyIndex.scala

```diff
@@ -153,19 +196,81 @@ class PartKeyTantivyIndex(ref: DatasetRef,
   }
 
   override protected def addIndexedField(key: String, value: String): Unit = {
-    ???
+    val buffer = docBufferLocal.get()
+
+    // 1 - indexed field
+    buffer += 1
+    TantivyQueryBuilder.writeStringToBuffer(key, buffer)
+    TantivyQueryBuilder.writeStringToBuffer(value, buffer)
   }
 
   protected def addIndexedMapField(mapColumn: String, key: String, value: String): Unit = {
-    ???
+    val buffer = docBufferLocal.get()
+
+    // 2 - map field
+    buffer += 2
+    TantivyQueryBuilder.writeStringToBuffer(mapColumn, buffer)
```

Comment:
Will utf8 strings work correctly with this approach of writing from Java and reading from rust ?
A unit test would be good if it does not exist

Reply:
Yes, the existing tests exercise this including non-ASCII values.

They're not running yet because they need the query side implemented as well, but they'll get merged in as part of this PR series.